### PR TITLE
[#5401] Support -Djdk.tls.ephemeralDHKeySize=num when using OpenSslCo…

### PR DESCRIPTION
**Pull request by normanmaurer**
…ntext

Motivation:

Java8+ adds support set a DH key size via a System property (jdk.tls.ephemeralDHKeySize). We should respect this when using OpenSSL.

Modifications:

Respect system property.

Result:

More consistent SSL implementation.

### CODE REVIEW

**Comment by normanmaurer** [comment]
This depends on https://github.com/netty/netty-tcnative/pull/162

File: handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java [hunk 1]
```diff
@@ -121,6 +124,23 @@ public SelectedListenerFailureBehavior selectedListenerFailureBehavior() {
         if (logger.isDebugEnabled()) {
             logger.debug("Default cipher suite (OpenSSL): " + ciphers);
         }
+
+        String dhKeySize = AccessController.doPrivileged(new PrivilegedAction<String>() {
```
Comment (Scottmitch) [inline handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java]: put this in a `catch (Throwable t)` block?

File: handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java [hunk 2]
```diff
@@ -121,6 +124,29 @@ public SelectedListenerFailureBehavior selectedListenerFailureBehavior() {
...
+        try {
+            String dhKeySize = AccessController.doPrivileged(new PrivilegedAction<String>() {
+                @Override
+                public String run() {
+                    return SystemPropertyUtil.get("jdk.tls.ephemeralDHKeySize");
+                }
+            });
+            if (dhKeySize != null) {
+                try {
+                    dhLen = Integer.parseInt(dhKeySize);
```
Comment (netkins) [inline handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java]: MAJOR Method io.netty.handler.ssl.OpenSslContext.<static initializer for OpenSslContext>() converts String to boxed primitive using excessive boxing [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=fb-contrib%3ANAB_NEEDLESS_BOXING_VALUEOF)

File: handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java [hunk 3]
```diff
@@ -121,6 +124,29 @@ public SelectedListenerFailureBehavior selectedListenerFailureBehavior() {
...
+
+        try {
+            String dhKeySize = AccessController.doPrivileged(new PrivilegedAction<String>() {
+                @Override
+                public String run() {
+                    return SystemPropertyUtil.get("jdk.tls.ephemeralDHKeySize");
+                }
+            });
+            if (dhKeySize != null) {
+                try {
```
Comment (netkins) [inline handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java]: MAJOR Extract this nested try block into a separate method. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1141)

File: handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java [hunk 4]
```diff
@@ -121,6 +124,29 @@ public SelectedListenerFailureBehavior selectedListenerFailureBehavior() {
...
+                    dhLen = Integer.parseInt(dhKeySize);
+                } catch (NumberFormatException e) {
+                    logger.debug("OpenSslContext only support -Djdk.tls.ephemeralDHKeySize={int}, but got: "
+                            + dhKeySize);
+                    dhLen = null;
```
Comment (Scottmitch) [inline handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java]: is this necessary?

Comment (normanmaurer) [inline handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java]: nope let me remove it.

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (3a69adfefb6136e5c63727d389b784b921dee08c) and 4.0 (7077840)

**Pull request closed without merging**